components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/AnalyticsPublisher.java

```diff
@@ -0,0 +1,273 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.bpmn.analytics.publisher;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.bpmn.analytics.publisher.internal.BPMNAnalyticsHolder;
+import org.wso2.carbon.bpmn.analytics.publisher.models.BPMNProcessInstance;
+import org.wso2.carbon.bpmn.analytics.publisher.models.BPMNTaskInstance;
+import org.wso2.carbon.bpmn.analytics.publisher.utils.AnalyticsPublishServiceUtils;
+import org.wso2.carbon.bpmn.analytics.publisher.utils.BPMNDataReceiverConfig;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.context.RegistryType;
+import org.wso2.carbon.databridge.agent.thrift.DataPublisher;
+import org.wso2.carbon.databridge.agent.thrift.exception.AgentException;
+import org.wso2.carbon.databridge.commons.AttributeType;
+import org.wso2.carbon.databridge.commons.StreamDefinition;
+import org.wso2.carbon.databridge.commons.exception.*;
+import org.wso2.carbon.registry.api.Registry;
+import org.wso2.carbon.registry.api.RegistryException;
+import org.wso2.carbon.registry.core.utils.RegistryUtils;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.concurrent.Executors;
+
+/**
+ * AnalyticsPublisher uses to publish events to the data receiver in data-bridge
+ */
+public class AnalyticsPublisher {
+	private static Log log = LogFactory.getLog(AnalyticsPublisher.class);
+
+	private String processInstanceStreamId;
+	private String taskInstanceStreamId;
+	private DataPublisher dataPublisher;
+	private AnalyticsPublishServiceUtils analyticsPublishServiceUtils;
+
+	/**
+	 * Initialize the objects for AnalyticsPublisher
+	 */
+	public void initialize() {
+		try {
+			RegistryUtils.setTrustStoreSystemProperties();
+			dataPublisher = createDataPublisher();
+			processInstanceStreamId = getBPMNProcessInstanceStreamId();
+			taskInstanceStreamId = getBPMNTaskInstanceStreamId();
+			analyticsPublishServiceUtils = new AnalyticsPublishServiceUtils();
+			int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+			String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+			Registry registry = BPMNAnalyticsHolder.getInstance().getRegistryService()
+			                                       .getGovernanceSystemRegistry();
+			setPrivilegeContext(tenantId, tenantDomain, registry);
+		} catch (MalformedURLException | AgentException | AuthenticationException | TransportException |
+				DifferentStreamDefinitionAlreadyDefinedException | StreamDefinitionException |
+				MalformedStreamDefinitionException | UserStoreException | RegistryException e) {
+			String errMsg = "Data publisher objects initialization error.";
+			log.error(errMsg, e);
```

Comment:
Is there any specific reason for not throwing an exception from here? Shouldn't initialize() operation fail in an exception case?